src/interfaces/IKintoApp.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+interface IKintoApp {
+    /* ============ Structs ============ */
+
+    struct Metadata {
```

Comment:
not sure if it's that relevant on L2, but just because i'm used to, some potential gas optis:
- variable packing:
 ```suggestion
struct Metadata {
    address developerWallet; // 160 bits
    bool dsaEnabled;         // 1 bit + padding
    uint128 rateLimitPeriod;
    uint128 rateLimitNumber;
    uint256 gasLimitPeriod;
    uint256 gasLimitCost;
    string name;
}
```
i guess we don't need uint256 for the rate limits?
`uint128 rateLimitPeriod;`
 `uint128 rateLimitNumber;`
 i suggested 128 but up to you...

Reply:
yeah, these are always good. In an L2, instead of storage being expensive, calldata is what needs to be optimized more because it is what gets propagated to L1.

I'll change it

Reply:
hmm, modified the order only because the gasLimitCost is eth so it needs 256 and I pass them all as array of uint256

Reply:
> yeah, these are always good. In an L2, instead of storage being expensive, calldata is what needs to be optimized more because it is what gets propagated to L1.

ohh, good to know!